pkg/cluster/registerWithHive.go

```diff
@@ -0,0 +1,127 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"os"
+	"time"
+
+	hive "github.com/Azure/ARO-RP/pkg/hiveregistrator"
+	"github.com/openshift/installer/pkg/asset/installconfig/azure"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (m *manager) collectDataForHive(ctx context.Context) error {
+
+	m.log.Print("collecting registration data from new cluster")
+
+	cv, err := m.configcli.ConfigV1().ClusterVersions().Get(ctx, "version", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	clusterid := cv.Spec.ClusterID
+
+	servicePrincipal := azure.Credentials{
+		TenantID:       m.env.TenantID(),
+		SubscriptionID: m.env.SubscriptionID(),
+		ClientID:       m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.ClientID,
+		ClientSecret:   string(m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile.ClientSecret),
+	}
+
+	spBytes, err := json.Marshal(servicePrincipal)
+	if err != nil {
+		return err
+	}
+
+	// TODO: What if a customer adds/modifies a pullsecret after the install ?
+	// TODO: All parameter locations to be verified
+	m.hiveWorkloadCluster = &hive.WorkloadCluster{
```

Comment:
Are we setting it on manager because we need to share the state between actions? If so, maybe we should follow existing pattern:

https://github.com/Azure/ARO-RP/blob/51fd449e7e6054938388f29f64f412944d31ca15/pkg/cluster/install.go#L128-L131

https://github.com/Azure/ARO-RP/blob/51fd449e7e6054938388f29f64f412944d31ca15/pkg/cluster/install.go#L139-L143

https://github.com/Azure/ARO-RP/blob/51fd449e7e6054938388f29f64f412944d31ca15/pkg/cluster/install.go#L147-L149